### PR TITLE
feat(CI/Jenkins): Add parameter type checking/mapped elements to Jenkins task UI.

### PR DESCRIPTION
**Pull request by Atchuu**
Changes were made to the Jenkins task UI used by Deck to support
specific parameter types already provided by Igor. The default was left as
input:text to keep the functionality for unknown/unspecified parameter types.

Recommendations/changes welcome.

[https://github.com/spinnaker/spinnaker/issues/1872](Issue)

### CODE REVIEW

File: app/scripts/modules/core/src/pipeline/config/stages/jenkins/jenkinsStage.html [hunk 1]
```diff
@@ -65,9 +65,38 @@ <h4 class="text-left" >Job Parameters</h4>
...
-        <input disabled ng-if="useDefaultParameters[parameter.name]" type="text" class="form-control input-sm" value="{{parameter.defaultValue}}"/>
-        <input ng-if="!useDefaultParameters[parameter.name]" type="text" class="form-control input-sm" ng-model="userSuppliedParameters[parameter.name]" ng-change="updateParam(parameter.name)"/>
+      <div class="col-md-5" ng-switch on="parameter.type">
+        <div ng-switch-when="PasswordParameterDefinition">
+          <!-- Passwords do not expose their defaults, for obvious reasons, so we can skip this -->
+          <input ng-if="!useDefaultParameters[parameter.name]" type="password" class="form-control input-sm" ng-model="userSuppliedParameters[parameter.name]" ng-change="updateParam(parameter.name)"/>
+        </div>
+        <div ng-switch-when="BooleanParameterDefinition">
+          <input disabled ng-if="useDefaultParameters[parameter.name] && parameter.defaultValue == 'true'" type="checkbox" class="input-sm" checked/>
+          <input disabled ng-if="useDefaultParameters[parameter.name] && parameter.defaultValue == 'false'" type="checkbox" class="input-sm"/>
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/stages/jenkins/jenkinsStage.html]: Can these be collapsed into a single input?
```html
<input disabled ng-if="useDefaultParameters[parameter.name] type="checkbox" ng-check="parameter.defaultValue === 'true'"/>
```
?

Comment (Atchuu) [inline app/scripts/modules/core/src/pipeline/config/stages/jenkins/jenkinsStage.html]: I will test it and find out. Give me one second.

Comment (Atchuu) [inline app/scripts/modules/core/src/pipeline/config/stages/jenkins/jenkinsStage.html]: It doesn't appear to like it. I seem to recall trying this when I was first setting the boolean section up.

image

Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/stages/jenkins/jenkinsStage.html]: Cool, thanks for checking

Comment (Atchuu) [inline app/scripts/modules/core/src/pipeline/config/stages/jenkins/jenkinsStage.html]: There was a syntax error that I didn't notice (missing "). Fixing it allows the checkbox to appear. The problem now is that default value seems to be ignored. (it isn't auto-checking the checkbox) I can investigate further if you think it's useful, but at this point, it's one extra if/then check so I'm not sure if it's a good use of time :). Thoughts?

**Pull request merged**